### PR TITLE
[WIP] Manifest based webhooks implementation

### PR DESCRIPTION
**Pull request by vivekbagade**
**What type of PR is this?**

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

https://github.com/kubernetes/enhancements/issues/1872

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

File: pkg/kubeapiserver/admission/BUILD [hunk 1]
```diff
@@ -5,15 +5,22 @@ go_library(
     srcs = [
         "config.go",
         "initializer.go",
+        "webhook_defaulting_validation.go",
     ],
     importpath = "k8s.io/kubernetes/pkg/kubeapiserver/admission",
     visibility = ["//visibility:public"],
     deps = [
+        "//pkg/apis/admissionregistration:go_default_library",
```
Comment (donmeganotz) [inline pkg/kubeapiserver/admission/BUILD:13]: image
```suggestion
     srcs = [
         "config.go",
         "initializer.go",
        "webhook_defaulting_validation.go",
     ],
     importpath = "k8s.io/kubernetes/pkg/kubeapiserver/admission",
     visibility = ["//visibility:public"],
     deps = [
#        "//pkg/apis/admissionregistration:go_default_library",
```

**Review by donmeganotz — CHANGES_REQUESTED** [review]

File: pkg/kubeapiserver/admission/initializer.go [hunk 2]
```diff
@@ -40,11 +41,19 @@ type WantsQuotaConfiguration interface {
 	admission.InitializationValidator
 }
 
+// WantsDefaulterAndValidator defines a function which sets the validator and defaulter for admission plugins
+// that need it.
+type WantsDefaulterAndValidator interface {
+	SetDefaulterAndValidator(d generic.WebhookDefaulter, v generic.WebhookValidator)
+}
+
 // PluginInitializer is used for initialization of the Kubernetes specific admission plugins.
```
Comment (donmeganotz) [inline pkg/kubeapiserver/admission/initializer.go:43]: Oic@

**Review by donmeganotz — APPROVED** [review]

**Review by donmeganotz — CHANGES_REQUESTED** [review]
**](( @@megate ott```

```

**Comment by liggitt** [comment]
closing per https://github.com/kubernetes/enhancements/issues/1872#issuecomment-767158504

/close

**Pull request closed without merging**